### ISSUE TITLE
HMR not working in non-React examples

### ISSUE THREAD

**Issue by Hypnosphi** [storybooks/storybook#3454]
### Steps to reproduce

Open vue, angular, polymer, or mithril example locally, change something in the story. Storybook doesn't get updated without page reload. This only reproduces on `master` branch, not on `release/3.4`
HMR works only in React

cc @igor-dv

**Comment by Hypnosphi** [comment]
There's a chance it's somehow related to warnings during compilation. I'm adding graphql example to storybook-official, it has compilation warning and breaks HMR. I noticed that angular compilation has warnings as well:

```
WARNING in /Users/jetbrains/IdeaProjects/storybook/node_modules/@angular/core/esm5/core.js
System.import() is deprecated and will be removed soon. Use import() instead.
For more info visit https://webpack.js.org/guides/code-splitting/
 @ ./src/stories/custom-ng-content.stories.ts 6570:15-36 12:13-37
 @ ./src/stories sync \.stories\.ts$
 @ ./.storybook/config.js
 @ multi /Users/jetbrains/IdeaProjects/storybook/app/angular/dist/server/config/polyfills.js /Users/jetbrains/IdeaProjects/storybook/app/angular/dist/server/config/globals.js (webpack)-hot-middleware/client.js?reload=true ./.storybook/config.js

WARNING in /Users/jetbrains/IdeaProjects/storybook/node_modules/@angular/core/esm5/core.js
System.import() is deprecated and will be removed soon. Use import() instead.
For more info visit https://webpack.js.org/guides/code-splitting/
 @ ./src/stories/custom-ng-content.stories.ts 6590:15-102 12:13-37
 @ ./src/stories sync \.stories\.ts$
 @ ./.storybook/config.js
 @ multi /Users/jetbrains/IdeaProjects/storybook/app/angular/dist/server/config/polyfills.js /Users/jetbrains/IdeaProjects/storybook/app/angular/dist/server/config/globals.js (webpack)-hot-middleware/client.js?reload=true ./.storybook/config.js

```

**Comment by igor-dv** [comment]
Oh, I am glad that was the issue =)

**Comment by Hypnosphi** [comment]
Released as [`4.0.0-alpha.4`](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.4)
